text_generation/causal_lm/cpp/causal_lm.cpp

```diff
@@ -39,13 +39,10 @@ int main(int argc, char* argv[]) try {
         }},
         {1, ov::PartialShape{
             BATCH_SIZE, -1
-        }},
-        {2, ov::PartialShape{
-	    BATCH_SIZE, -1
         }}
     };
     std::vector<ov::Output<ov::Node>> inputs = model->inputs();
-    for (size_t idx = 3; idx < inputs.size(); ++idx) {
+    for (size_t idx = 2; idx < inputs.size(); ++idx) {
         ov::PartialShape shape = inputs.at(idx).get_partial_shape();
         shape[0] = BATCH_SIZE;
```

Comment:
as far as I remember, chatglm has different index for batch dimension

Reply:
Yes, this PR is only for llama, will add chatglm support in another PR.

Reply:
Apparently exporting a model as stateful removes any difference between the dimensions. I was able to run chatglm2-6b and chatglm3-6bit using https://github.com/Wovchena/openvino.genai-public/pull/8 and get meaningful output. Here's the PR updating convert.py: https://github.com/openvinotoolkit/openvino.genai/pull/52. My next step is to rewrite this causal_lm as stateful as well, so we could decide if we need stateless approach at all.

Reply:
I agree.
But we should wait until stateful models support is merged to HF, right?
It would be great to avoid when we refer to different commits to convert different models

Reply:
We are considering to have separate samples for stateless models if they demonstrate popular architectures and require architecture-dependent kv-cache processing.

Reply:
But the main focus is on stateful models.

Reply:
so, maybe we can merge chatglm sample as is and treat it as "sample for stateless model for popular architecture" ?
While current one will show stateful models and hence it become more generic

Reply:
Greedy search with stateful model: https://github.com/sammysun0711/openvino.genai/pull/1 That eliminates a need to have special implementation for chatglm.

I also had that Idea to have stateless model with architecture, which doesn't fit into the general scenario. And there's still Qwen which may require that https://github.com/openvinotoolkit/openvino.genai/pull/43. I haven't tried it yet